crates/biome_js_analyze/src/lint/nursery/use_index_of.rs

```diff
@@ -0,0 +1,363 @@
+use crate::JsRuleAction;
+use biome_analyze::{
+    Ast, FixKind, Rule, RuleDiagnostic, RuleSource, context::RuleContext, declare_lint_rule,
+};
+use biome_console::markup;
+use biome_diagnostics::Severity;
+use biome_js_syntax::{
+    AnyJsArrowFunctionParameters, AnyJsCallArgument, AnyJsExpression, AnyJsFunctionBody,
+    AnyJsMemberExpression, JsArrowFunctionExpression, JsAssignmentExpression, JsBinaryExpression,
+    JsCallExpression, JsFunctionExpression, JsLogicalExpression, JsParameterList,
+    JsReturnStatement, JsSyntaxNode, JsSyntaxToken, JsVariableDeclaration, T,
+};
+use biome_rowan::{AstNode, AstSeparatedList, BatchMutationExt, SyntaxToken};
+
+pub struct JsSyntaxMatchPair {
+    pub member_name: JsSyntaxToken,
+    pub matching_array_element: JsSyntaxNode,
+}
+
+fn member_name_remap(old_member_name: &str) -> Option<&str> {
+    let new_member_name = match old_member_name {
+        "findIndex" => "indexOf",
+        "findLastIndex" => "lastIndexOf",
+        _ => return None,
+    };
+    Some(new_member_name)
+}
+
+fn extract_simple_compare_match(
+    expression: &JsBinaryExpression,
+    parameter_name: &String,
+) -> Option<JsSyntaxNode> {
+    if expression.operator_token().ok()?.kind() != T![===] {
+        return None;
+    }
+
+    let (left, right) = (expression.left().ok()?, expression.right().ok()?);
+
+    let matching_side = if left.syntax().to_string().trim() == *parameter_name {
```

Comment:
I am on mobile so couldn't check, but there is something like this if I remember correctly:

```suggestion
    let matching_side = if left.syntax().to_text_trimmed() == *parameter_name {
```